kolibri/core/discovery/utils/network/search.py

```diff
@@ -130,31 +145,75 @@ def remove_service(self, zeroconf, type, name):
             del self.instances[id]
 
 
-def get_available_instances(timeout=2, include_local=True):
+cache = Cache(os.path.join(KOLIBRI_HOME, "zeroconf_cache"))
```

Comment:
using `diskcache` for now, although it might be nice to use redis if it's available (to avoid writes on raspberry pi sd cards, for example.

Reply:
Also for Cloud Kolibri we have seen issues when the underlying file system is a virtualised POSIX file system on top of a Windows file system.

My currently open PR #6195 has a similar issue - I think we can probably address both with a compatability layer similar to the cross process cache that uses Redis when available.

Reply:
Looking at the way you are using it, I think the cross process cache wrapper that is already implemented would handle this? (it will use redis when available, but otherwise will default to the `diskcache` based caching layer).